packages/core/src/components/NumberInput.tsx

```diff
@@ -8,68 +8,67 @@ interface Props {
   onChangeText?: (value?: number) => void;
 }
 
-const NumberInput: FC<Props> = ({
-  onChangeText,
-  value,
-  defaultValue,
-  ...props
-}) => {
-  const [currentStringNumberValue, setCurrentStringNumberValue] = useState("");
+const NumberInput = React.forwardRef<TextInput, Props>(
```

Comment:
Changes in this file are mostly indentations. Main change is `React.forwardRef`, and passing that ref to the `TextInput`